### PR TITLE
refactor: Python packaging

### LINKED ISSUE #11: Make `colorizer-data` writer scripts into a Python package
Allow scientists to easily import the package and use it with their existing data pipelines. This also helps transfer ownership of file conversion to the projects instead of this central repo.

It may also be possible for us to set up the github repo so it can be directly imported via pip.

### PR DESCRIPTION
**Pull request by ShrimpCryptid**
Closes #11 !

- Refactors `data_writer_utils`, which has been split into a `writer` and `utils` module within the new `colorizer_data` package.
- Updates documentation with instructions on installing and updating the project.

Testing:
1. Create a clean conda environment.
2. Run `pip install git+https://github.com/allen-cell-animated/colorizer-data.git@refactor/python-package`.
3. Run `python` to open the interactive environment.
4. Paste in the following lines. It should run without errors.
```
from colorizer_data.writer import (
    ColorizerDatasetWriter,
)
from colorizer_data.utils import generate_frame_paths

writer = ColorizerDatasetWriter(".", "", 1)
print(writer)

print(generate_frame_paths(34))
```

### CODE REVIEW

File: colorizer_data/utils.py
Comment (ShrimpCryptid) [inline colorizer_data/utils.py:1]: Copied directly from `data_writer_utils.py`.

File: README.md
Comment (toloudis) [inline README.md:11]: Nit:  `conda activate` is actually a separate concern from having access to on-prem Vast storage.  Using a Python virtual environment is a general best practice for any python code.  You might word this like "make sure you have read access to your source data and write access to your output directory!"

File: README.md
Comment (toloudis) [inline README.md:25]: Consider making this the default instruction up above with the current version, when we version it for the first time.

File: README.md
Comment (toloudis) [inline README.md:117]: please add the dry run option here. You should do a dry run first to make sure the changes are as expected.

File: colorizer_data/utils.py
Comment (toloudis) [reply colorizer_data/utils.py:1]: which is now writer.py ?

File: README.md
Comment (toloudis) [reply README.md:117]: also, please give an explicit example so that people know what the proper form of [version], [type], and [new tag] should be.   Specifically, the tags need to begin with `v`.

Comment (toloudis) [reply README.md:117]: Because of the multi-file configuration options, you can have the latest version in this README file be updated also by the same script. (Referencing my comment above about how users should install from latest tagged version)

File: pyproject.toml
Comment (toloudis) [inline pyproject.toml:12]: I think you can add bump-my-version as a dev dependency in here

File: colorizer_data/utils.py
Comment (ShrimpCryptid) [reply colorizer_data/utils.py:1]: yes!

File: README.md
Comment (ShrimpCryptid) [reply README.md]: FYI, everything is set to 0.0.0 currently! Once this PR is merged, I'll do the first major version bump to take us to 1.0.0.

File: README.md
Comment (ShrimpCryptid) [reply README.md]: > also, please give an explicit example so that people know what the proper form of [version], [type], and [new tag] should be. Specifically, the tags need to begin with `v`.

Yup! There's already an example down below, though I've updated it and the instructions.

File: pyproject.toml
Comment (ShrimpCryptid) [reply pyproject.toml:12]: I'm going to leave it out of the dependencies, because it'll require anyone who installs `colorizer-data` to also install it. Scientists won't really need it, and any devs who want to contribute should already be reading our README and following the install step there, I hope.

Comment (toloudis) [reply pyproject.toml:12]: The way to formalize it is like this:
```
[project.optional-dependencies]
# development dependency groups
dev = [
    'bump-my-version',
]
```
Then a contributor can do `pip install -e .[dev]`

Comment (toloudis) [reply pyproject.toml:12]: But I know it's only one single package so no big deal yet

File: README.md
Comment (toloudis) [inline README.md:15]: I still think this makes it seem like the user needs to know what version they want ahead of time.  The first thing under installation should just be a one-liner to get current latest version and make it clear that's what they will get.  After that, you can give instructions on how to install arbitrary versions.

Suggestion:
```
Installation: 

# pip
pip install git+https://github.com/allen-cell-animated/colorizer-data.git@v0.0.0
# requirements.txt
colorizer-data @ git+https://github.com/allen-cell-animated/colorizer-data.git@v0.0.0

To install a different version, replace the last part of the url with the version or tag of your choice such as `@vX.X.X` or `@{branch-name}`

File: README.md
Comment (toloudis) [inline README.md:116]: I think there might even be a simpler command like `git push origin --tags`.

File: .bumpversion.toml
Comment (toloudis) [inline .bumpversion.toml:1]: 👍

File: README.md
Comment (ShrimpCryptid) [reply README.md:116]: Interesting note on this! I saw on StackOverflow that developers are discouraged from running `git push origin --tags`, because it can also push potentially unwanted tags from other local development.

The two alternatives were pushing the specific tag name (current approach), or pushing with the `--folow-tags` option, which ignores unannotated or unreachable tags. I slightly prefer having devs push specific tag names as a validation step... what do you think?

Comment (toloudis) [reply README.md:120]: Makes sense.  I only make tags for versioning/releasing on main branch that I know should be in the origin remote.  So I would never have this issue.  But it's true you could be pushing local tags that would be meaningless and pollute the repo for others.  As long as you're aware of what it means.

File: pyproject.toml
Comment (toloudis) [inline pyproject.toml:48]: In general, if the example scripts are installed with this package, then their dependencies should be dependencies here.  I'm not sure if there is a normal counterexample for this, but you need to at least document why after installing this package, a bin script would fail.  Maybe either in the README or a comment at the top of the example scripts.

Comment (toloudis) [reply pyproject.toml:48]: something like "WARNING: the example scripts include external dependencies which you will have to install manually as they are not considered proper dependencies of this package"

**Review by toloudis — APPROVED** [review]
I think if you just add a blurb about the example scripts dependences, then this is looking good

File: pyproject.toml
Comment (ShrimpCryptid) [reply pyproject.toml:48]: Done, thank you!

**Pull request merged**
